src/process-hooks.ts

```diff
@@ -23,12 +28,19 @@ export default (tagFunction: () => TramOneElement) => {
 	const existingEffects = getEffectStore(TRAM_EFFECT_STORE);
 	const queuedEffects = getEffectStore(TRAM_EFFECT_QUEUE);
 
+	// get all new keys
+	const newKeys = getKeyQueue(TRAM_KEY_QUEUE);
+
 	// store new effects in the node we just built
 	const newEffects = Object.keys(queuedEffects).filter((effect) => !(effect in existingEffects));
 	tagResult[TRAM_TAG_NEW_EFFECTS] = newEffects.map((newEffectKey) => queuedEffects[newEffectKey]);
 
-	// restore the effect queue to what it was before we started
+	// store keys in the node we just built
+	tagResult[TRAM_TAG_STORE_KEYS] = newKeys;
```

Comment:
this is what associates the observable keys with the element. Basically, after we've generated all the elements (and called any store hooks), we put those keys on the node, which can then be picked up by the mutation observer.